docs/external-db.md

```diff
@@ -0,0 +1,161 @@
+# External DB integration
+
+Backstage hosts the data in a [PostgreSQL database](https://backstage.io/docs/getting-started/config/database/).
+By default, the Operator creates and manages a local instance of PostgreSQL in the same namespace as the Backstage deployment but it also allows to switch this off and configure an external database server instead.
+Usually, external connection requires more security, so, this instruction includes steps to configure SSL/TLS.
+
+### Configure your external PostgreSQL instance
+As a prerequisite, you have to know:
+- **db-host** - your PostgreSQL instance DNS or IP address 
+- **db-port** - your PostgreSQL instance port number (usually 5432)
+- **username** - to connect to your PostgreSQL instance
+- **password** - to connect to your PostgreSQL instance
+
+**NOTE:** By default, Backstage uses databases for each plugin and automatically creates them if none are found, so in addition to PSQL Database level privileges, the user may need Create Database privilege.  
+
+In addition, to get your database connection secured with SSL/TLS, you also need certificates in the form of PEM file. 
+
+You can find configuration guidelines for:
+- [AWS RDS PostgreSQL](#aws-rds-postgresql)
+- [Azure Database PostgreSQL](#azure-db-postgresql)
+
+### Create secret with PostgreSQL connection properties:
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <cred-secret-name>
+type: Opaque
+stringData:
+ POSTGRES_PASSWORD: <password>
+ POSTGRES_PORT: <db-port>
+ POSTGRES_USER: <username>
+ POSTGRES_HOST: <db-host>
+ PGSSLMODE: require #  for TLS connection
+ NODE_EXTRA_CA_CERTS: <abs-path-to-pem-file> # for TLS connection, e.g. /opt/app-root/src/postgres-crt.pem
```

Comment:
Per https://backstage.io/docs/tutorials/switching-sqlite-postgres/#add-postgresql-configuration, there is a `backend.database.connection.ssl` object in the app-config, where we can configure the `node-postgres` client with things like a client certificate file for example (besides a CA cert file) - see https://node-postgres.com/features/ssl.
Did you test this path instead?
Maybe I've missed something, but I am not sure specifying a client certificate is do-able via the `NODE_EXTRA_CA_CERTS` env var (it covers CA certs, I guess).

Reply:
nvm. Per https://node-postgres.com/features/connecting, it seems the `node-postgres` client honors the same [environment variables](https://www.postgresql.org/docs/current/libpq-envars.html#LIBPQ-ENVARS) as `psql` to connect to a PostgreSQL server. So in the case of a client cert, I guess we can pass a `PGSSLCERT` env var.
But maybe it is worth mentioning somewhere that those environment variables should work?

Reply:
I seens, yes :) but it did not work for me, show me please if it works for you.